packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -364,6 +362,116 @@ export function getMiniflareOptions(
 	};
 }
 
+export function getPreviewMiniflareOptions(
+	normalizedPluginConfig: NormalizedPluginConfig,
+	viteConfig: vite.ResolvedConfig,
+): MiniflareOptions {
+	const entryWorkerConfig = normalizedPluginConfig.entryWorkerName
+		? normalizedPluginConfig.workers[normalizedPluginConfig.entryWorkerName]
+		: undefined;
+
+	const assetsDirectory = path.resolve(
+		viteConfig.root,
+		viteConfig.build.outDir,
+		'client',
+	);
+	const hasAssets = fs.existsSync(assetsDirectory);
+	const assets = hasAssets
+		? {
+				assets: {
+					routingConfig: {
+						has_user_worker: entryWorkerConfig ? true : false,
+					},
+					assetConfig: {
+						...(normalizedPluginConfig.assets.htmlHandling
+							? { html_handling: normalizedPluginConfig.assets.htmlHandling }
+							: {}),
+						...(normalizedPluginConfig.assets.notFoundHandling
+							? {
+									not_found_handling:
+										normalizedPluginConfig.assets.notFoundHandling,
+								}
+							: {}),
+					},
+					directory: assetsDirectory,
+					...(entryWorkerConfig?.assetsBinding
+						? { binding: entryWorkerConfig.assetsBinding }
+						: {}),
+				},
+			}
+		: {};
+
+	const workers: Array<WorkerOptions> = [
+		...(entryWorkerConfig
+			? [
+					{
+						...entryWorkerConfig.workerOptions,
+						...assets,
+						modules: [
+							{
+								type: 'ESModule',
+								path: path.resolve(
+									viteConfig.root,
+									viteConfig.build.outDir,
+									entryWorkerConfig.workerOptions.name,
+									'index.js',
+								),
+							} as const,
+						],
+					},
+				]
+			: [
+					{
+						...assets,
+						name: 'assets-only',
+						modules: [
+							{
+								type: 'ESModule',
+								path: '__dummy-module__',
+								contents: '',
+							} as const,
+						],
```

Comment:
I think this can be simplified
```suggestion
						script: '',
						modules: true,
```

Reply:
b4f921a